### PR TITLE
[Feat] Create Filter Box Component

### PR DESCRIPTION
**Pull request by helenellyx**
## Notion ticket link

[Filter Function - Filter Boxes Component](https://www.notion.so/uwblueprintexecs/Filter-Function-Filter-Boxes-Component-6df327e0766947de894a776cdc409665)

## Implementation description
* Note that the design of the filter box is not included in this ticket
* Created the filter box component with two filters applied
* audience: single select filter
* genres: multi-select filter

## Steps to test
1. Instead of passing options to choose from in the dropdown menu of both filters as a prop, hardcode two arrays of options and test with your arrays 
2. verify that the audience is single-select
3. verify that genres is multi-select
4. verify that `genres`  and `age` has the correct value 
5. verify that when clear is clicked, both `genres` and `age` are cleared and the filter box is cleared
6. **note**, the only way to test the apply button is to check that `genres`  and `age` has the correct value, since right now the genres and age filters that the user chose is passed to a function outside of the component which has not been defined 

## What should reviewers focus on?
* the test route is `/test/filter`
* please check that I did not break any functionalities for the genre and tags selection for adding a book

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/FilterBox.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            label=""
+            placeholder="Genres"
+            options={genreOptions}
+            setOptions={null}
+            optionsSelected={genres}
+            setOptionsSelected={setGenres}
+            allowMultiSelectOption
+            allowAddOption={false}
+          />
+          <Button onClick={clickApply}>Apply</Button>
```
Comment (jessiiepeng) [inline frontend/src/components/pages/FilterBox.tsx:61]: I think these two "Apply" and "Clear" can be centered vertically of the boxes: they're a little more towards the top right now.
Screen Shot 2022-08-03 at 7 17 03 PM

**Comment by jessiiepeng** [comment]
Do we need to make this component responsive in this ticket? In mobile view it looks a bit squished the words
Screen Shot 2022-08-03 at 7 22 48 PM

File: frontend/src/components/pages/FilterBox.tsx (hunk 1 above)
Comment (jessiiepeng) [reply frontend/src/components/pages/FilterBox.tsx:61]: Do we need to make this responsive for mobile view? Looks a bit squished right now
Screen Shot 2022-08-03 at 7 23 32 PM
.

**Comment by jessiiepeng** [comment]
The functionality looks good, I don't think it broke anything for adding tags/genres on books. Just some small styling fixes.

File: frontend/src/components/pages/FilterBox.tsx (hunk 1 above)
Comment (helenellyx) [reply frontend/src/components/pages/FilterBox.tsx:61]: I think @susheelk mentioned that the design of the filter boxes and buttons is not included in this ticket 🤔 I could work on it tho if needed

Comment (susheelk) [reply frontend/src/components/pages/FilterBox.tsx:61]: I think it's a good idea to make this somewhat responsive, even if it doesn't have to look good. Maybe have the buttons flow to a new line?

**Review by jessiiepeng — APPROVED** [review]
LGTM! yay :)

**Pull request merged**